File: HW2/Animal.cpp
#include "Animal.h"


const char *Animal::food_source_names_[] =
        {"CARNIVORE",
         "HERBIVORE",
         "INSECTIVORE",
         "OMNIVORE"};

Animal::Animal(std::string *name, int weight, FoodSource foodSource) {
    name_ = *name;
    weight_ = weight;
    foodSource_ = foodSource;
}

Animal *Animal::clone() {
    return new Animal(&name_, weight_, foodSource_);
}

std::string Animal::toString() {
    return "ANIMAL { name = \"" + name_ + "\"; " +
           "weight = " + std::to_string(weight_) + "; " +
           "food_source = " + food_source_names_[foodSource_ - 1] +
           "; metric = " + std::to_string(getMetric()) + " }";
}

File: HW3/Animal.py
from Creature import Creature


# The animal type of creatures.
class Animal(Creature):
    food_sources = ["CARNIVORE", "HERBIVORE", "INSECTIVORE", "OMNIVORE"]

    def __init__(self, name: str, weight: int, food_source: int):
        super().__init__(name, weight)
        if 1 <= food_source <= len(self.food_sources):
            self.food_source = self.food_sources[food_source - 1]
        else:
            raise Exception(f"Invalid food source for animal {name}")

    def to_string(self) -> str:
        return f"ANIMAL {{ name = \"{self.name}\"; " \
               f"weight = {self.weight}; "\
               f"food_source = {self.food_source}; "\
               f"metric = {self.get_metric()} }}"
